citesphere/src/main/java/edu/asu/diging/citesphere/core/service/impl/GroupManager.java

```diff
@@ -43,4 +48,12 @@ public ICitationGroup getGroup(IUser user, String groupId) {
         }
         return null;
     }
+    
+    @Override
+    public void deleteLocalGroupCopy(String groupId) throws GroupDoesNotExistException {
+        Long deleteCount = customGroupRepo.deleteByGroupId(Integer.parseInt(groupId));
+        if (deleteCount == 0L) {
+            throw new GroupDoesNotExistException("There is no group with id " + groupId);
```

Comment:
I wouldn't throw an exception here. It's more of a philosophy thing, but if something that doesn't exist is being deleted, the outcome is still correct (the "something" doesn't exist after the action).